x/auth/keeper/deterministic_test.go

```diff
@@ -118,3 +124,425 @@ func pubkeyGenerator(t *rapid.T) *rapid.Generator[secp256k1.PubKey] {
+			suite.Require().Equal(res.GetInfo().AccountNumber, prevRes.GetAccountNumber())
+			suite.Require().Equal(res.GetInfo().Sequence, prevRes.Sequence)
+		}
+
+		prevRes = res.GetInfo()
+	}
+}
+
+func (suite *DeterministicTestSuite) TestGRPCQueryAccountInfo() {
+	rapid.Check(suite.T(), func(t *rapid.T) {
+		pub := pubkeyGenerator(t).Draw(t, "pubkey")
+		addr := sdk.AccAddress(pub.Address())
+		accNum := rapid.Uint64().Draw(t, "account-number")
+		seq := rapid.Uint64().Draw(t, "sequence")
+
+		acc1 := types.NewBaseAccount(addr, &pub, accNum, seq)
+		suite.accountKeeper.SetAccount(suite.ctx, acc1)
+
+		suite.runAccountInfoIterations(addr, acc1)
+	})
+
+	// Regression test
+	addr1 := sdk.MustAccAddressFromBech32("cosmos1j364pjm8jkxxmujj0vp2xjg0y7w8tyveuamfm6")
+	pub, err := hex.DecodeString("01090C02812F010C25200ED40E004105160196E801F70005070EA21603FF06001E")
+	suite.Require().NoError(err)
+
+	accNum := uint64(10087)
+	seq := uint64(0)
+
+	acc1 := types.NewBaseAccount(addr1, &secp256k1.PubKey{Key: pub}, accNum, seq)
+
+	suite.accountKeeper.SetAccount(suite.ctx, acc1)
+	suite.runAccountInfoIterations(addr1, acc1)
+}
+
+func (suite *DeterministicTestSuite) createAndReturnQueryClient(ak keeper.AccountKeeper) types.QueryClient {
+	queryHelper := baseapp.NewQueryServerTestHelper(suite.ctx, suite.encCfg.InterfaceRegistry)
+	types.RegisterQueryServer(queryHelper, ak)
+	return types.NewQueryClient(queryHelper)
+}
+
+func (suite *DeterministicTestSuite) runBech32PrefixIterations(ak keeper.AccountKeeper, preRes string) {
+	queryClient := suite.createAndReturnQueryClient(ak)
+
+	for i := 0; i < 1000; i++ {
+		res, err := queryClient.Bech32Prefix(suite.ctx, &types.Bech32PrefixRequest{})
+		suite.Require().NoError(err)
+		suite.Require().NotNil(res)
+
+		suite.Require().Equal(res.Bech32Prefix, preRes)
+		preRes = res.Bech32Prefix
+	}
+}
+
+func (suite *DeterministicTestSuite) TestGRPCQueryBech32Prefix() {
+	rapid.Check(suite.T(), func(t *rapid.T) {
+		prefix := rapid.StringMatching(`[a-zA-Z]+[1-9a-zA-Z]*`).Draw(t, "prefix")
+		ak := keeper.NewAccountKeeper(
+			suite.encCfg.Codec,
+			suite.key,
+			types.ProtoBaseAccount,
+			nil,
+			prefix,
+			types.NewModuleAddress("gov").String(),
+		)
+
+		suite.runBech32PrefixIterations(ak, prefix)
+	})
+
+	prefix := "prefix"
+	ak := keeper.NewAccountKeeper(
+		suite.encCfg.Codec,
+		suite.key,
+		types.ProtoBaseAccount,
+		nil,
+		prefix,
+		types.NewModuleAddress("gov").String(),
+	)
+
+	suite.runBech32PrefixIterations(ak, prefix)
+}
+
+func (suite *DeterministicTestSuite) runAddressBytesToStringIterations(addressBytes []byte, prevRes string) {
+	for i := 0; i < 1000; i++ {
+		res, err := suite.queryClient.AddressBytesToString(suite.ctx, &types.AddressBytesToStringRequest{
+			AddressBytes: addressBytes,
+		})
+
+		suite.Require().NoError(err)
+		suite.Require().NotNil(res)
+
+		suite.Require().Equal(res.AddressString, prevRes)
+		prevRes = res.AddressString
```

Comment:
```suggestion
```

same for other tests?

Reply:
isn't it required to store the last response?

Reply:
but you have it in input params ...?

Reply:
Since the last response is equal to the input param (as per the suite.Equal just above), there's no need to re-store the last response.